From 6cc925329f22bb3b7a4beefadb2e7e73450a594b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rainer=20M=C3=BCller?= <raimue@codingfarm.de>
Date: Wed, 20 Feb 2013 18:31:48 +0100
Subject: [PATCH 1/4] configure: check for program_invocation_name

For portabiliy, check for program_invocation_name during configure and
define HAVE_PROGRAM_INVOCATION_NAME accordingly. Use of this symbol is
now enclosed with the appropriate #ifdef block.

The symbol program_invocation_name is only used for error message
handling using error(), so it's safe to omit this if it is not
available.
---
 configure.ac | 8 ++++++++
 free.c       | 4 +++-
 pgrep.c      | 2 ++
 pmap.c       | 4 +++-
 pwdx.c       | 4 +++-
 skill.c      | 4 +++-
 slabtop.c    | 2 ++
 sysctl.c     | 2 ++
 tload.c      | 2 ++
 uptime.c     | 2 ++
 vmstat.c     | 2 ++
 w.c          | 2 ++
 watch.c      | 2 ++
 13 files changed, 36 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 711be2a..afb24ce 100644
--- a/configure.ac
+++ b/configure.ac
@@ -137,6 +137,14 @@ dnl else
 dnl	ALL_LINGUAS="af am ar as be bg bn_IN bn ca cs cy da de el en_GB es et eu_ES fa fi fr gl gu he hi hr hu hy id is it ja ka kn ko ku lo lt lv mk ml mr ms my nb nl nn no nso or pa pl pt_BR pt ro ru si sk sl sq sr@Latn sr sv ta te th tr uk ur vi zh_CN zh_TW zu"
 dnl fi
 
+AC_MSG_CHECKING(whether program_invocation_name is defined)
+AC_TRY_COMPILE([#include <errno.h>],
+		[program_invocation_name = "test";],
+		AC_DEFINE(HAVE_PROGRAM_INVOCATION_NAME, 1,
+			[Define if program_invocation_name is defined])
+		AC_MSG_RESULT(yes),
+		AC_MSG_RESULT(no))
+
 AC_MSG_CHECKING(whether program_invocation_short_name is defined)
 AC_TRY_COMPILE([#include <errno.h>],
 		[program_invocation_short_name = "test";],
diff --git a/free.c b/free.c
index ecaeccd..ffc4b86 100644
--- a/free.c
+++ b/free.c
@@ -223,7 +223,9 @@ int main(int argc, char **argv)
 	args.repeat_interval = 1000000;
 	args.repeat_counter = 0;
 
-    program_invocation_name = program_invocation_short_name;
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
+	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale (LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
diff --git a/pgrep.c b/pgrep.c
index f08d89f..e849f9e 100644
--- a/pgrep.c
+++ b/pgrep.c
@@ -788,7 +788,9 @@ int main (int argc, char **argv)
 	struct el *procs;
 	int num;
 
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
 	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale (LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
diff --git a/pmap.c b/pmap.c
index 79c17b2..fd0363b 100644
--- a/pmap.c
+++ b/pmap.c
@@ -657,7 +657,9 @@ int main(int argc, char **argv)
 		{NULL, 0, NULL, 0}
 	};
 
-    program_invocation_name = program_invocation_short_name;
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
+	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale (LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
diff --git a/pwdx.c b/pwdx.c
index 4aee389..214275a 100644
--- a/pwdx.c
+++ b/pwdx.c
@@ -74,7 +74,9 @@ int main(int argc, char *argv[])
 		{NULL, 0, 0, 0}
 	};
 
-    program_invocation_name = program_invocation_short_name;
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
+	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale (LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
diff --git a/skill.c b/skill.c
index 03df229..fb57305 100644
--- a/skill.c
+++ b/skill.c
@@ -629,7 +629,9 @@ static void skillsnice_parse(int argc,
 /* main body */
 int main(int argc, char ** argv)
 {
-    program_invocation_name = program_invocation_short_name;
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
+	program_invocation_name = program_invocation_short_name;
+#endif
 	struct run_time_conf_t run_time;
 	memset(&run_time, 0, sizeof(struct run_time_conf_t));
 	my_pid = getpid();
diff --git a/slabtop.c b/slabtop.c
index 9e535d4..ebdae8d 100644
--- a/slabtop.c
+++ b/slabtop.c
@@ -300,7 +300,9 @@ int main(int argc, char *argv[])
 		{  NULL, 0, NULL, 0 }
 	};
 
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
 	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale (LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
diff --git a/sysctl.c b/sysctl.c
index 0756878..2eeacd9 100644
--- a/sysctl.c
+++ b/sysctl.c
@@ -675,7 +675,9 @@ int main(int argc, char *argv[])
 		{NULL, 0, NULL, 0}
 	};
 
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
 	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
diff --git a/tload.c b/tload.c
index 765ed17..a5fe14a 100644
--- a/tload.c
+++ b/tload.c
@@ -114,7 +114,9 @@ int main(int argc, char **argv)
 		{NULL, 0, NULL, 0}
 	};
  
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
 	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale (LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
diff --git a/uptime.c b/uptime.c
index ac4579c..2742189 100644
--- a/uptime.c
+++ b/uptime.c
@@ -77,7 +77,9 @@ int main(int argc, char **argv)
 		{NULL, 0, NULL, 0}
 	};
 
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
 	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale (LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
diff --git a/vmstat.c b/vmstat.c
index a2627b8..6f95f09 100644
--- a/vmstat.c
+++ b/vmstat.c
@@ -724,7 +724,9 @@ int main(int argc, char *argv[])
 		{NULL, 0, NULL, 0}
 	};
 
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
 	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale (LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
diff --git a/w.c b/w.c
index 6352d69..5db1064 100644
--- a/w.c
+++ b/w.c
@@ -483,7 +483,9 @@ int main(int argc, char **argv)
 		{NULL, 0, NULL, 0}
 	};
 
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
 	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale (LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
diff --git a/watch.c b/watch.c
index 0fe0a8f..5a22952 100644
--- a/watch.c
+++ b/watch.c
@@ -590,7 +590,9 @@ int main(int argc, char *argv[])
 		{0, 0, 0, 0}
 	};
 
+#ifdef HAVE_PROGRAM_INVOCATION_NAME
 	program_invocation_name = program_invocation_short_name;
+#endif
 	setlocale(LC_ALL, "");
 	bindtextdomain(PACKAGE, LOCALEDIR);
 	textdomain(PACKAGE);
-- 
1.8.1.3

